src/face_op.cpp

```diff
@@ -271,33 +271,42 @@ void Manifold::Impl::FlattenFaces() {
            [](auto& v) { v.store(0); });
 
   for_each_n(policy, countAt(0_uz), numTri,
-             [&edgeFace, &vertDegree, this](size_t tri) {
+             [&edgeTri, &vertDegree, this](size_t tri) {
+               const auto& ref = meshRelation_.triRef[tri];
                for (const int i : {0, 1, 2}) {
                  const int edge = 3 * tri + i;
                  const int pair = halfedge_[edge].pairedHalfedge;
                  if (pair < 0) {
-                   edgeFace[edge] = kRemove;
+                   edgeTri[edge] = kRemove;
                    return;
                  }
-                 const auto& ref = meshRelation_.triRef[tri];
                  if (ref.SameFace(meshRelation_.triRef[pair / 3])) {
-                   edgeFace[edge] = kRemove;
+                   edgeTri[edge] = kRemove;
                  } else {
-                   edgeFace[edge] = (static_cast<uint64_t>(ref.meshID) << 32) +
-                                    static_cast<uint64_t>(ref.coplanarID);
+                   edgeTri[edge] = tri;
                    ++vertDegree[halfedge_[edge].startVert];
                  }
                }
              });
 
+  auto comp = [&edgeTri, this](size_t a, size_t b) {
+    if (edgeTri[a] == edgeTri[b]) return false;
+    if (edgeTri[a] == kRemove) return false;
+    if (edgeTri[b] == kRemove) return true;
+    const TriRef& refA = meshRelation_.triRef[edgeTri[a]];
+    const TriRef& refB = meshRelation_.triRef[edgeTri[b]];
+    return refA.meshID < refB.meshID || (refA.meshID == refB.meshID &&
```

Comment:
One neat trick I just found is that we can use `std::tie` to make a tuple and a tuple comparison to do lexicographic comparisons.